src/pp_25_guessing_game_two.py

```diff
@@ -0,0 +1,96 @@
+import random
+
+
+def input_validator(input_number: int) -> bool:
+    """
+    user guess validator
+
+    Args:
+        input_guess (int): user input integer
+
+    Returns:
+        bool: approval of the validation
+    """
+    if 1 <= input_number <= 9:
+        return True
+
+    return False
+
+
+def guess_the_number(input_number: int, guess_number: int) -> dict:
+    """
+    Args:
+        input_number (int): user input number
+        guess_number (int): by programe guess number
+
+    Returns:
+        dict: validation
+    """
+    if not input_validator(input_number):
+        print("You have to give me the number between 1 and 9 (included). Try again!")
+        return False
+
+    return evaluate_guess(input_number, guess_number)
+
+
+def evaluate_guess(user_input: int, guess_number: int) -> dict:
+    """
+    guess and input number evaluation
+
+    Args:
+        user_input (int): user input number
+        guess_number (int): by programme guessed number
+
+    Returns:
+        dict: dictionary with min and max value
+    """
+    min_num = 1
+    max_num = 9
+
+    if user_input == guess_number:
+        print("The number is {number}. You won!".format(number=user_input))
+        return {
+            'min' : min_num,
+            'max' : max_num
+            }
+
+    if guess_number < user_input:
+        min_num = guess_number + 1
+        print("Your guess {guess} is lower than the number. Guess again".format(guess=guess_number))
+    elif guess_number > user_input:
+        max_num = guess_number - 1
+        print("Your guess {guess} is higher than the number. Guess again".format(
+            guess=guess_number))
+
+    return {
+        'min' : min_num,
+        'max' : max_num
+        }
+
+
+def guess_game_two():
+    """
+    starter of the guess game two
+    """
+    user_input = int(input("Give me the number between 1 and 9 (included): "))
+    print(user_input)
+    number_of_guesses = 0
+    min_num = 1
+    max_num = 9
+    while True:
+        guess_number = random.randint(min_num, max_num)
+        print("Program guessed number: {guess_number}.".format(guess_number=guess_number))
+        game_state = guess_the_number(user_input, guess_number)
```

Comment:
First of all `guess_the_number` method is pointles.

Secondly, I think you might not following the acceptance criteria here.
> The program will guess a number, and you, the user, will say whether it is too high, too low, or your number.

- It means, we should do the logic, we have above. 
- Then it print's the number , computer guessed - it should be a random number, but there should be some logic behind it. - line 81
- User should insert `too low`, `too high` or `right on the money`
- Program should do another loop, that this user's input and according to this value it should "guess" a different number, which is hopefully closer to the result.  

I would be interesting what "guessing algorythm" you can come up with. In case, you need a hint, just ping me :)

Reply:
So at the end the terminal output shoule look like this:

```
$ python3 src/pp_25_guessing_game_two.py
Give me your secret number:  #actually this doesn't have to be here as far as you keep the number in your mind.
6 # user input

I'm guessing that the secret number is 4 Am I right?
higher # user input
I'm guessing that the secret number is 7 Am I right?
lower # user input
I'm guessing that the secret number is 5 Am I right?
yes # user input

🎉 Cha! I've told you. First try!
```